### PR TITLE
Handling warnings and adding OIDC promotion message release PR

### LINKED ISSUE #216: Error `Unable to encode the output with cp1252 encoding.` is shown when logging in to a subscription which name contains Japanese characters and on Windows runner
When trying to log in with a subscription which name contains Japanese characters by the following simple workflow, the logging in is failed with the encoding error like the following screenshot.

Workflow:
```yaml
name: Azure Login Test

on:
  workflow_dispatch:

jobs:
  login:
    runs-on: windows-latest # using Windows runner
    steps:
    - uses: azure/login@v1
      with:
        creds: ${{ secrets.AZURE_CREDENTIALS }}
```

Error screenshot:
image

### PR DESCRIPTION
**Pull request by BALAGA-GAYATRI**
Resolves #216 
- Warnings will not be thrown as errors
- Adding an additional OIDC promotion message when a user opts for Non-OIDC login
 Screenshot 2022-05-02 at 12 46 27 PM

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -12,22 +12,20 @@ var azPSHostEnv = !!process.env.AZUREPS_HOST_ENVIRONMENT ? `${process.env.AZUREP
...
-                    //removing the keyword 'ERROR' to avoid duplicates while throwing error
-                    if (error.toLowerCase().startsWith('error')) {
-                        error = error.slice(5);
-                    }
-                    // printing error
-                    if (error && error.trim().length !== 0) {
-                        commandStdErr = true;
-                        core.error(error);
+                    let isWarning = error.toLowerCase().startsWith('warning');
+                    let isError = error.toLowerCase().startsWith('error');
```
Comment (tauhid621) [inline src/main.ts:21]: NIT : Can be renamed with `startsWithError`.

File: src/main.ts [hunk 2]
```diff
@@ -12,22 +12,20 @@ var azPSHostEnv = !!process.env.AZUREPS_HOST_ENVIRONMENT ? `${process.env.AZUREP
...
+                    // printing ERROR
+                    if (error && error.trim().length !== 0 && !isWarning) {
+                        if(isError) {
+                            //removing the keyword 'ERROR' to avoid duplicates while throwing error
+                            error = error.slice(5);
+                        }
+                        core.setFailed(error);
```
Comment (tauhid621) [inline src/main.ts:28]: Earlier it was `core.error(error)`. What is the difference?

**Review by tauhid621 — APPROVED** [review]
Looks Good. Left NIT comments.

File: src/main.ts (hunk 2 above)
Comment (BALAGA-GAYATRI) [reply src/main.ts:28]: The latter is used to set the exit codes and mark them as failed. In our case both are the same. That shouldn't be a problem.

File: src/main.ts (hunk 1 above)
Comment (BALAGA-GAYATRI) [reply src/main.ts]: Done 👍

**Pull request merged**